modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+                                                                    Object objectNEWUser,
+                                                                    Object objectOLDUser,
+                                                                    int expectedScimResponseStatus)
+            throws BadRequestException, CharonException, NotFoundException, NotImplementedException {
+
+        User userNew = (User) objectNEWUser;
+        User userOld = (User) objectOLDUser;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        Mockito.when(userManager.getUser(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
+        User validatedUser = (User) ServerSideValidator.validateUpdatedSCIMObject(userOld, userNew, schema);
+        Mockito.when(userManager.updateUser(any(User.class), anyMap(), anyList())).thenReturn(validatedUser);
+        SCIMResponse scimResponse = userResourceManager.updateWithPATCH(existingId, scimObjectString,
+                null, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHNotFoundException")
+    public Object[][] dataToUpdateWithPATCHNotFoundException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName",
+                        null, userNew, userOld, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCHNotFoundException")
+    public void testUpdateWithPATCHNoAssociatedUserExitsInTheUserStore(String existingId, String scimObjectString,
+                                                                       String attributes, String excludeAttributes,
+                                                                       Object objectNEWUser,
+                                                                       Object objectOLDUser,
```

Comment:
```suggestion
                                                                       Object objectNewUser,
                                                                       Object objectOldUser,
```

Reply:
merge this into 1 1line

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11